### PR TITLE
QoS Over Gateway

### PR DESCRIPTION
**Pull request by derrandz**

### CODE REVIEW

File: package.json
Comment (derrandz) [inline package.json:6]: niet

File: src/datasource/gateway.js
Comment (derrandz) [inline src/datasource/gateway.js:87]: Tracking this as an [issue](https://github.com/pokt-foundation/portal-api/issues/111)

File: src/datasource/gateway.js
Comment (derrandz) [inline src/datasource/gateway.js:136]: I think `_functionName` is confusing, I will switch to:

```
 requestHandlers = {
    handlerName
  }
 ```
 
 If you are wondering why at all, it is to allow for easier interception pre and post request. Check the following lines in `query = `

File: src/datasource/gateway.test.js
Comment (derrandz) [inline src/datasource/gateway.test.js:22]: This was hardcoded in here for convenience's sake. I will add a bit of `||` to allow for using `.env.test` if you anyone fancies that. Will prove useful in CI to use env variables in instead.

File: src/datasource/gateway.test.js
Comment (derrandz) [inline src/datasource/gateway.test.js:134]: This will allow for proper abstraction, to make tests configurable. Can help in CI too.

File: src/views/account-detail/index.js
Comment (derrandz) [inline src/views/account-detail/index.js:227]: This is old, it should go away by the time I rebase this branch.

Comment (derrandz) [reply src/views/account-detail/index.js:227]: For the record, I did not write this LOL

File: x
Comment (Evalir) [inline x:1]: ohhh what's this haha?

File: src/views/transaction-detail/index.js
Comment (Evalir) [inline src/views/transaction-detail/index.js:148]: Interesting—shouldn't we do this asynchronously?

EDIT: Ah, interesting, I think this is okay.

File: src/views/transaction-detail/index.js
Comment (Evalir) [inline src/views/transaction-detail/index.js:18]: Again with the parenthesis :).

File: src/views/transaction-detail/index.js
Comment (Evalir) [inline src/views/transaction-detail/index.js:28]: Oof, this could probably be null or give an error, depending on how the URL is copied. Can't we destructure and assign this in `componentWillMount`? Feels like something _can_ explode here.

File: src/views/send/index.js
Comment (Evalir) [inline src/views/send/index.js:171]: What happens when you get pushed to this page? The tx info probably won't be able to be fetched as the block has not passed, so, wonder what's the expected behavior?

File: src/datasource/gateway.js
Comment (Evalir) [inline src/datasource/gateway.js:192]: Are we sure we wanna destructure `response`? Thinking we might not know what's inside of this object at all times.

File: src/datasource/gateway.js
Comment (Evalir) [reply src/datasource/gateway.js:136]: Definitely appreciate this change :)

**Review by Evalir — APPROVED** [review]
Wasn't as thorough of a review as I would've liked, but for the code we have, this LGTM. Just comments and some extreme nitpicking as always :).

File: .env
Comment (nymd) [inline .env:26]: blockchain-subdomain has been removed, please use actual subdomain

Comment (nymd) [reply .env:26]: this header may be blocked by CORS and throw an error

File: src/core/services/pocket-service.js
Comment (nymd) [reply src/core/services/pocket-service.js:2]: A different PR please

File: src/datasource/gateway-test-utils.js
Comment (nymd) [reply src/datasource/gateway-test-utils.js:28]: I need it to work at this point, can be prettier later, thanks

File: src/datasource/gateway.js
Comment (nymd) [reply src/datasource/gateway.js:87]: Not an issue, see issue

File: src/views/transaction-detail/index.js
Comment (nymd) [reply src/views/transaction-detail/index.js:28]: If this is all old code, we can ignore for now

**Review by nymd — APPROVED** [review]

File: src/core/services/pocket-service.js
Comment (derrandz) [reply src/core/services/pocket-service.js:2]: Prettier is not setup but I am sure this one is due to me running a globally installed one.

File: x
Comment (derrandz) [reply x:1]: That's an intruder!

File: src/views/transaction-detail/index.js
Comment (derrandz) [reply src/views/transaction-detail/index.js:148]: Not my doing per se, just a fix of what was there already

File: src/datasource/gateway.js
Comment (derrandz) [reply src/datasource/gateway.js:192]: We sorta do that's why

File: .env
Comment (derrandz) [reply .env:26]: sounds good, this was a hack for local dev thanks for catching

Comment (derrandz) [reply .env:26]: actually this is double catch .env should not be committed so 👍🏻

Comment (derrandz) [reply .env:26]: and in workflows env variables there is no blockchain subdomain

**Comment by derrandz** [comment]
All prettiness stuff I've highlighted and you guys have highlighted are for not taking's sake. Merging until a later time.

**Pull request merged**
